docs/docs/architecture.md

```diff
@@ -170,59 +175,62 @@ sequenceDiagram
 
 ![sequence diagram](images/sequence.png)
 
-The first time SGX is used on a machine there are interactions with either the cloud provider or Intel to retrieve
-machine certificates proving authenticity. The host then gets a remote attestation ([`EnclaveInstanceInfo`](api/-conclave%20-core/com.r3.conclave.common/-enclave-instance-info/index.html)) to the
-client somehow, the client verifies it and optionally asks Intel if the hardware setup of the machine is still
-considered to be secure, or if there are known vulnerabilities (see [renewability](renewability.md)). This can be
-repeated as often as the client wants, e.g. every day. Once this is done the client can send messages to the enclave
-through the host.
+1. When you use SGX in a machine for the first time, the host interacts with the cloud provider or Intel to retrieve
+   certificates to prove authenticity.
+2. The host then sends a remote attestation ([`EnclaveInstanceInfo`](api/-conclave%20-core/com.r3.conclave.common/-enclave-instance-info/index.html))
+   to the client.
+3. The client verifies the remote attestation and optionally asks Intel if the hardware setup of the machine is still
+   secure or if there are known vulnerabilities (see [renewability](renewability.md)). This step can be repeated as 
+   often as the client wants.
+4. The client communicates to the enclave through the host by using encrypted messages and responses.
 
-### Mail
+### Conclave Mail
 
-Communicating with an enclave requires sending and receiving encrypted and authenticated messages. One possible approach
-is to embed a TLS stack into the enclave and use something like HTTPS, but this technique has some problems
-and limitations that are resolved via [the Conclave Mail API](mail.md). Mail makes communication between enclaves and
-clients easy, as well as solving common problems faced with doing application development. 
+Clients must securely communicate with enclaves without the host spying on the messages. Conclave uses
+[Conclave Mail](mail.md) to send and receive encrypted messages between a client and an enclave. Conclave Mail makes 
```

Comment:
I think we need to make the link to the mail page more explicit, something like "Learn more about Conclave Mail here"

Reply:
done.